src/domain/request/RequestUrlParser.js

```diff
@@ -3,7 +3,7 @@ const { InvalidUrlError, InvalidContentTypeError } = require('../common/utils/er
 
 class RequestUrlParser {
   static resolveRawUrl(wrappedUrl) {
-    const urlRegex = new RegExp(/\(https?:\/\/.+\)/);
+    const urlRegex = new RegExp(/\(https?:\/\/[^)]+\)/);
```

Comment:
I'm not sure if I understand your intention here but I think this change is not needed and may actually be unsafe. '.+' operator is greedy by default, so it will select whole query correctly even if closing braces are part of the query (something like this http://prntscr.com/o3cbtt) . I believe this is correct behavior, as closing brace is a legal character that could be a part of http URL.

Reply:
This change was needed according to successfully support query like this: `xml(http://samples.openweathermap.org/data/2.5/weather?q=London&mode=xml&appid=b6907d289e10d714a6e88b30761fae22)string(/current/temperature/@value)`. Previously this query breaks flow as the link was fetched as `http://samples.openweathermap.org/data/2.5/weather?q=London&mode=xml&appid=b6907d289e10d714a6e88b30761fae22)string(/current/temperature/@value` and not `http://samples.openweathermap.org/data/2.5/weather?q=London&mode=xml&appid=b6907d289e10d714a6e88b30761fae22`

Reply:
Fair enough.